src/tutorials/lua_basics_lightbulb.md

```diff
@@ -365,398 +390,227 @@ switchTrigger.interactedEvent:Connect(OnSwitchInteraction)
 
-Your `OnSwitchInteraction()` function should now look like this:
+Your `OnTriggerInteracted()` function should now look like this:
 
 ```lua
-local function OnSwitchInteraction()
-    if not lightIsOn then
-        switch:RotateTo(onRotation, 0.5, true)
-        World.SpawnAsset(propLightTemplate, {position = bulbPosition})
+function OnTriggerInteracted()
+    -- Flips the cached state of if the switch is on or off
+    isSwitchOn = not isSwitchOn
+
+    if isSwitchOn then
+        Switch:RotateTo(ROTATION_ON, TIME_ROTATE, true)
+        Light.visibility = Visibility.INHERIT
     else
-        switch:RotateTo(offRotation, 0.5, true)
+        Switch:RotateTo(ROTATION_OFF, TIME_ROTATE, true)
     end
-
-    lightIsOn = not lightIsOn
 end
 ```
 
-Press **Play** to see if your `else` statement works. The switch should now rotate up when first interacted with, then down on your second interaction with it. However, we still need to de-spawn (delete) the light when the light switch is turned off.
-
----
+Press **Play** to see if your `else` statement works. The switch should now rotate up when first interacted with, then down on your second interaction with it. However, we still need to turn off the light when the light switch is turned off.
 
 ## Turning the Light Off
 
-### Make a Variable for the Spawned Light
-
-In order to turn off the light, you first need to define the light after it is spawned. When the **LightTemplate** is spawned, it shows up at the bottom of the Hierarchy. Make a variable to reference this at the top of the script:
-
-```lua
-local spawnedLight = nil
-```
+Your next project goal is to make the switch more functional and have it turn on a light when players interact with it.
 
-- `spawnedLight` is the name you are giving to the light that spawns.
-- `nil` means that it has no value yet.
-
-Next, in the line where you spawn the light, assign what is spawned to the variable, by adding `spawnedLight =` to the beginning:
+In the `OnTriggerInteracted()` in the else statement under the `RotateTo()` function, add:
 
 ```lua
-    spawnedLight = World.SpawnAsset(propLightTemplate, {position = bulbPosition})
+Light.visibility = Visibility.FORCE_OFF
 ```
 
-- `World.SpawnAsset` is doing two different things here. It creates the asset in the game space, and also **returns** the unique ID of the asset that just got spawned, which you are saving to the variable `spawnedLight`.
-
-### Destroy the Light
+- `FORCE_FF` disables visibility.
 
-Now that you have defined our spawned light, you can tell the script to destroy it when the switch is turned off. In the **else** section, add:
+The OnTriggerInteracted function should look like:
 
 ```lua
-spawnedLight:Destroy()
-```
-
-- `spawnedLight` is the variable we just defined representing the Point Light spawned when the light switch is turned on.
-- `Destroy()` is a function used to delete objects from a scene.
+-- Function to change the switch state and turn on/off the like
```

Comment:
"turn the light on and off", plus it changes the the variable that keeps track of whether it is on or off, AND rotates the switch. Two confusingly similar concepts that mean totally different things.